crates/core/tests/client.rs

```diff
@@ -278,6 +292,9 @@ mod tests {
         let balance =
             client.balance(ethereum_address_to_deploy, BlockId::Number(BlockNumberOrTag::Latest)).await.unwrap();
         assert_eq!(balance, U256::ZERO);
+
+        let nonce = client.nonce(ethereum_address_to_deploy, BlockId::Number(BlockNumberOrTag::Latest)).await.unwrap();
+        assert_eq!(nonce, U256::ZERO);
```

Comment:
not sure if I should make a separate `test_get_nonce_eoa` so I just added  the `nonce` check here